src/components/RoadmapList/RoadmapList.tsx

```diff
@@ -2,56 +2,96 @@ import { useEffect, useState } from 'react';
+const DEFAULT_FILTER_VALUES = {
+  category: '',
+  level: '',
+  domain: '',
+  search: '',
+};
 
 const RoadmapList = () => {
   const [currentPage, setCurrentPage] = useState(1);
   const [totalPage, setTotalPage] = useState(1);
   const [list, setList] = useState<RoadmapListItem[]>([]);
+  const [filters, setFilters] = useState(DEFAULT_FILTER_VALUES);
 
   useEffect(() => {
-    getRoadmapList(currentPage).then(({ data }) => {
-      setList(data.results);
-      setTotalPage(Math.ceil(data.count / ROADMAPS_PER_PAGE));
-    });
-  }, [currentPage]);
+    if (filters === DEFAULT_FILTER_VALUES) {
+      getRoadmapList(currentPage).then(({ data }) => {
+        setList(data.results);
+        setTotalPage(Math.ceil(data.count / ROADMAPS_PER_PAGE));
+      });
+    }
+  }, [currentPage, filters]);
 
   const handlePageChange = ({ selected }: { selected: number }) => {
     setCurrentPage(selected + 1);
+    if (filters !== DEFAULT_FILTER_VALUES) {
+      handleFiltration(selected + 1)
+    }
+  };
+
+  const handleSelect = (name: string, value: string) => {
+    setFilters((state) => ({
+      ...state,
+      [name]: value,
+    }));
+  };
+
+  const handleSearch = () => {
+    handleFiltration()
+  };
+
+  const handleFiltration = (currentPage = 1)=>{
+    if (filters.search) {
+      getFiltredListWithSearch(filters, currentPage).then(({ data }) => {
+        setTotalPage(Math.ceil(data.count / ROADMAPS_PER_PAGE));
+        setList(data.results);
+      });
+      return;
+    }
+    if (!filters.search) {
+      getFiltredList(filters, currentPage).then(({ data }) => {
+        setTotalPage(Math.ceil(data.count / ROADMAPS_PER_PAGE));
+        setList(data.results);
+      });
+      return;
+    }
+  }
+
+  const handleFilterClear = () => {
+    setFilters(DEFAULT_FILTER_VALUES);
+    setCurrentPage(1)
+    getRoadmapList(1).then(({ data }) => {
+      setList(data.results);
+      setTotalPage(Math.ceil(data.count / ROADMAPS_PER_PAGE));
+    });
   };
 
   return (
     <div>
-      <div className={style.filterWrapper}>
-        <Container>
-          <p className={style.filterTitle}>Filters</p>
-          <div className={style.filterBox}>
-            <div className={style.filterList}>
-              <p className={style.filter}>Category</p>
-              <p className={style.filter}>Level</p>
-              <p className={style.filter}>Local domain</p>
-            </div>
-            <Button variant="primary" size="lg">
-              Search
-            </Button>
-          </div>
-        </Container>
-      </div>
+      <Filters onSelect={handleSelect} filters={filters} onSearch={handleSearch} onClear={handleFilterClear}/>
       <Container>
         <div className={style.listWrapper}>
-          {list.map((el) => (
-            <div key={el.pk} className={style.listItem}>
+          {list.map((el, ind) => (
+            <div key={`${el.pk}+${ind}`} className={style.listItem}>
               <div className={style.itemHeader}>
```

Comment:
why you add index here in key?

Reply:
Because the backend returns duplicate data that causes an error. I add an extra value to the key to make it unique